src/pynwb/legacy/map.py

```diff
@@ -25,54 +25,93 @@ class TypeMapLegacy(TypeMap):
-            if parent_ndt == 'Epoch':
-                return 'EpochTimeSeries'
-            if parent_ndt == 'MotionCorrection':
-                return 'CorrectedImageStack'
-            if parent_ndt == 'ImagingPlane' and isinstance(builder, GroupBuilder):
-                return 'OpticalChannel'
-            if parent_ndt == 'ImageSegmentation':
-                if builder.name in ('roi_ids', 'cell_specimen_ids'):
+
+            attrs = builder.attributes
+            ndt = attrs.get('neurodata_type')
+
+            if ndt == 'Module':
+                return 'ProcessingModule'
+            elif ndt == 'Interface':
+                return builder.name
+            elif ndt == 'Epoch':
+                return 'Epoch'
+            elif ndt == 'TimeSeries':
+                ancestry = decode(attrs['ancestry'][-1])
+                if ancestry == 'TwoPhotonSeries' and decode(builder.name) == 'corrected':
+                    return 'ImageSeries'
+                else:
+                    return ancestry
+            elif ndt == 'Custom':
+                parent_ndt = self.get_builder_dt(builder.parent)
+                if parent_ndt == 'ImageSegmentation' and builder.name in ('roi_ids', 'cell_specimen_ids'):
+                    return None
+                elif parent_ndt == 'PlaneSegmentation' and builder.name in ('roi_list', 'imaging_plane_name'):
+                    return None
+                elif parent_ndt == 'IntervalSeries' and builder.name in ('frame_duration',):
+                    return None
+                elif parent_ndt == 'TimeSeries' and builder.name in ('feature_description',
+                                                                     'bits_per_pixel',
+                                                                     'dimension',
+                                                                     'field_of_view',
+                                                                     'format'):
+                    return None
+                elif parent_ndt == 'ImagingPlane' and builder.parent.name in ('imaging_plane_1',):
+                    return None
+                elif parent_ndt == 'AbstractFeatureSeries' and builder.name in ('frame_duration',):
+                    return None
+                elif parent_ndt == 'IndexSeries' and builder.name in ('frame_duration',):
+                    return None
+                elif builder.parent.name == 'general':
+                    return None
+                elif parent_ndt == 'RoiResponseSeries' and builder.name in ('r',
+                                                                            'neuropil_traces_path',
+                                                                            'rmse',
+                                                                            'comments'):
                     return None
                 else:
-                    return 'PlaneSegmentation'
-
+                    raise Exception
```

Comment:
Avoid generic Exception. Choose from an existing specialized exception `RuntimeError` or similar . Make also sure to associate a descriptive message